Imports are lost when importing esm-like moduleswithout default export using commonjs import or esm import default

**Bug report**
Imports are lost when importing esm-like `export * from ...` modules without default export using commonjs import `import a = require( "mod" )` or esm import default `import a from "mod"`

**Input code**

example.ts
```ts
import repl1 = require( "node:repl" );
import repl2 from "node:repl";
import * as repl3 from "node:repl";

export interface MyReplOptions1 extends repl1.ReplOptions {}
export interface MyReplOptions2 extends repl2.ReplOptions {}
export interface MyReplOptions3 extends repl3.ReplOptions {}
```

tsconfig.json
```json
{
	"compilerOptions": {
		"target": "es2016",
		"module": "commonjs",
		"moduleResolution": "node",
		"declaration": true,
		"allowSyntheticDefaultImports": true,
		"esModuleInterop": true
	}
}
```

node_modules/@types/node/repl.d.ts (copy from `@types/node`@18.11.9)
```ts
declare module 'repl' {
    // some unimportant code(s) have been removed
    interface ReplOptions {
        // some unimportant code(s) have been removed
    }
    // some unimportant code(s) have been removed
}
declare module 'node:repl' {
    export * from 'repl';
}
```

command
```bash
dts-bundle-generator --project tsconfig.json example.ts
```

**Expected output**

example.d.ts
```ts
// Generated by dts-bundle-generator v7.1.0

import repl1 = require( "node:repl" );
import repl2 from "node:repl";
import * as repl3 from 'node:repl';

export interface MyReplOptions1 extends repl1.ReplOptions {
}
export interface MyReplOptions2 extends repl2.ReplOptions {
}
export interface MyReplOptions3 extends repl3.ReplOptions {
}

export {};
```

**Actual output**

example.d.ts
```ts
// Generated by dts-bundle-generator v7.1.0

import * as repl3 from 'node:repl';

export interface MyReplOptions1 extends repl1.ReplOptions {
}
export interface MyReplOptions2 extends repl2.ReplOptions {
}
export interface MyReplOptions3 extends repl3.ReplOptions {
}

export {};
```

command output
```
Compiling input files...
Processing example.ts
Writing example.ts -> example.d.ts
Checking generated files...
example.d.ts(5,41): error TS2833: Cannot find namespace 'repl1'. Did you mean 'repl3'?
example.d.ts(7,41): error TS2833: Cannot find namespace 'repl2'. Did you mean 'repl3'?

Error: Compiled with errors
```

**Additional context**
Add any other context about the problem here (CLI options, etc)
typescript 4.9.3 / dts-bundle-generator v7.1.0

--- a/src/bundle-generator.ts
+++ b/src/bundle-generator.ts
@@ -229,13 +229,17 @@ export function generateDtsBundle(entries: readonly EntryPointConfig[], options:
 
 				return leftSymbols.some((leftSymbol: ts.Symbol) => rightSymbols.includes(leftSymbol));
 			},
-			resolveReferencedModule: (node: ts.ExportDeclaration | ts.ModuleDeclaration | ts.ImportTypeNode) => {
+			resolveReferencedModule: (node: NodeWithReferencedModule) => {
 				let moduleName: ts.Expression | ts.LiteralTypeNode | undefined;
 
-				if (ts.isExportDeclaration(node)) {
+				if (ts.isExportDeclaration(node) || ts.isImportDeclaration(node)) {
 					moduleName = node.moduleSpecifier;
 				} else if (ts.isModuleDeclaration(node)) {
 					moduleName = node.name;
+				} else if (ts.isImportEqualsDeclaration(node)) {
+					if (ts.isExternalModuleReference(node.moduleReference)) {
+						moduleName = node.moduleReference.expression;
+					}
 				} else if (ts.isLiteralTypeNode(node.argument) && ts.isStringLiteral(node.argument.literal)) {
 					moduleName = node.argument.literal;
 				}
@@ -390,6 +394,8 @@ interface CollectingResult {
 	renamedExports: string[];
 }
 
+type NodeWithReferencedModule = ts.ExportDeclaration | ts.ModuleDeclaration | ts.ImportTypeNode | ts.ImportEqualsDeclaration | ts.ImportDeclaration;
+
 interface UpdateParams {
 	currentModule: ModuleInfo;
 	statements: readonly ts.Statement[];
@@ -406,7 +412,7 @@ interface UpdateParams {
 	getDeclarationsForExportedAssignment(exportAssignment: ts.ExportAssignment): ts.Declaration[];
 	getDeclarationUsagesSourceFiles(declaration: ts.NamedDeclaration): Set<ts.SourceFile | ts.ModuleDeclaration>;
 	areDeclarationSame(a: ts.NamedDeclaration, b: ts.NamedDeclaration): boolean;
-	resolveReferencedModule(node: ts.ExportDeclaration | ts.ModuleDeclaration | ts.ImportTypeNode): ts.SourceFile | ts.ModuleDeclaration | null;
+	resolveReferencedModule(node: NodeWithReferencedModule): ts.SourceFile | ts.ModuleDeclaration | null;
 }
 
 const skippedNodes = [
@@ -548,6 +554,19 @@ function updateResultForExternalEqExportAssignment(exportAssignment: ts.ExportAs
 	}
 }
 
+function getReferencedModuleInfo(moduleDecl: NodeWithReferencedModule, params: UpdateParams): ModuleInfo | null {
+	const referencedModule = params.resolveReferencedModule(moduleDecl);
+	if (referencedModule === null) {
+		return null;
+	}
+
+	const moduleFilePath = ts.isSourceFile(referencedModule)
+		? referencedModule.fileName
+		: resolveModuleFileName(referencedModule.getSourceFile().fileName, referencedModule.name.text);
+
+	return params.getModuleInfo(moduleFilePath);
+}
+
 function updateResultForModuleDeclaration(moduleDecl: ts.ModuleDeclaration, params: UpdateParams, result: CollectingResult): void {
 	if (moduleDecl.body === undefined || !ts.isModuleBlock(moduleDecl.body)) {
 		return;
@@ -559,16 +578,12 @@ function updateResultForModuleDeclaration(moduleDecl: ts.ModuleDeclaration, para
 		result.statements.push(moduleDecl);
 		return;
 	} else {
-		const referencedModule = params.resolveReferencedModule(moduleDecl);
-		if (referencedModule === null) {
+		const referencedModuleInfo = getReferencedModuleInfo(moduleDecl, params);
+		if (referencedModuleInfo === null) {
 			return;
 		}
 
-		const moduleFilePath = ts.isSourceFile(referencedModule)
-			? referencedModule.fileName
-			: resolveModuleFileName(referencedModule.getSourceFile().fileName, referencedModule.name.text);
-
-		moduleInfo = params.getModuleInfo(moduleFilePath);
+		moduleInfo = referencedModuleInfo;
 	}
 
 	// if we have declaration of external module inside internal one
@@ -676,23 +691,29 @@ function getImportModuleName(imp: ts.ImportEqualsDeclaration | ts.ImportDeclarat
 	return null;
 }
 
-function addImport(statement: ts.DeclarationStatement, params: UpdateParams, imports: CollectingResult['imports']): void {
-	if (statement.name === undefined) {
+function addImport(statement: ts.DeclarationStatement | ts.SourceFile, params: UpdateParams, imports: CollectingResult['imports']): void {
+	if (!ts.isSourceFile(statement) && statement.name === undefined) {
 		throw new Error(`Import/usage unnamed declaration: ${statement.getText()}`);
 	}
 
 	params.getDeclarationUsagesSourceFiles(statement).forEach((sourceFile: ts.SourceFile | ts.ModuleDeclaration) => {
-		const statements = ts.isSourceFile(sourceFile)
+		const sourceFileStatements = ts.isSourceFile(sourceFile)
 			? sourceFile.statements
 			: (sourceFile.body as ts.ModuleBlock).statements;
 
-		statements.forEach((st: ts.Statement) => {
+		sourceFileStatements.forEach((st: ts.Statement) => {
 			if (!ts.isImportEqualsDeclaration(st) && !ts.isImportDeclaration(st)) {
 				return;
 			}
 
 			const importModuleSpecifier = getImportModuleName(st);
 			if (importModuleSpecifier === null) {
+				return;
+			}
+
+			const referencedModuleInfo = getReferencedModuleInfo(st, params);
+			// if a referenced module should be inlined we can just ignore it
+			if (referencedModuleInfo === null || referencedModuleInfo.type !== ModuleType.ShouldBeImported) {
 				return;
 			}
 
@@ -769,7 +790,7 @@ function isNodeUsed(
 	typesUsageEvaluator: TypesUsageEvaluator,
 	typeChecker: ts.TypeChecker
 ): boolean {
-	if (isNodeNamedDeclaration(node)) {
+	if (isNodeNamedDeclaration(node) || ts.isSourceFile(node)) {
 		const nodeSymbol = getNodeSymbol(node, typeChecker);
 		if (nodeSymbol === null) {
 			return false;
@@ -859,6 +880,16 @@ function getExportedSymbolsUsingStatement(
 }
 
 function getNodeSymbol(node: ts.Node, typeChecker: ts.TypeChecker): ts.Symbol | null {
+	if (ts.isSourceFile(node)) {
+		const fileSymbol = typeChecker.getSymbolAtLocation(node);
+		// a source file might not have a symbol in case of no exports in that file
+		if (fileSymbol === undefined) {
+			return null;
+		}
+
+		return getActualSymbol(fileSymbol, typeChecker);
+	}
+
 	const nodeName = getNodeName(node);
 	if (nodeName === undefined) {
 		return null;

--- a/tests/e2e/test-cases/import-eq-with-interop/config.ts
+++ b/tests/e2e/test-cases/import-eq-with-interop/config.ts
@@ -1,0 +1,5 @@
+import { TestCaseConfig } from '../../test-cases/test-case-config';
+
+const config: TestCaseConfig = {};
+
+export = config;

--- a/tests/e2e/test-cases/import-eq-with-interop/export-eq.ts
+++ b/tests/e2e/test-cases/import-eq-with-interop/export-eq.ts
@@ -1,0 +1,3 @@
+class ClassName {}
+
+export = ClassName;

--- a/tests/e2e/test-cases/import-eq-with-interop/input.ts
+++ b/tests/e2e/test-cases/import-eq-with-interop/input.ts
@@ -1,0 +1,7 @@
+import MyModule from 'package-with-export-eq';
+import ClassName from './export-eq';
+
+export class ExportedClass extends ClassName implements MyModule.SomeCoolInterface {
+	public field: string = '';
+	public field2: number = 0;
+}

--- a/tests/e2e/test-cases/import-eq-with-interop/output.d.ts
+++ b/tests/e2e/test-cases/import-eq-with-interop/output.d.ts
@@ -1,0 +1,10 @@
+import MyModule from 'package-with-export-eq';
+
+declare class ClassName {
+}
+export declare class ExportedClass extends ClassName implements MyModule.SomeCoolInterface {
+	field: string;
+	field2: number;
+}
+
+export {};

--- a/tests/e2e/test-cases/import-eq-with-interop/tsconfig.json
+++ b/tests/e2e/test-cases/import-eq-with-interop/tsconfig.json
@@ -1,0 +1,7 @@
+{
+	"extends": "../tsconfig.json",
+	"compilerOptions": {
+		"allowSyntheticDefaultImports": true,
+		"esModuleInterop": true
+	}
+}

--- a/tests/e2e/test-cases/import-with-different-names/config.ts
+++ b/tests/e2e/test-cases/import-with-different-names/config.ts
@@ -1,0 +1,9 @@
+import { TestCaseConfig } from '../test-case-config';
+
+const config: TestCaseConfig = {
+	libraries: {
+		importedLibraries: ['fake-package', 'package-with-export-eq'],
+	},
+};
+
+export = config;

--- a/tests/e2e/test-cases/import-with-different-names/input.ts
+++ b/tests/e2e/test-cases/import-with-different-names/input.ts
@@ -1,0 +1,11 @@
+import fakePackage1 from 'fake-package';
+import * as fakePackage2 from 'fake-package';
+import fakePackage3 = require('fake-package');
+
+import { ExposedInterface } from './other-module';
+
+export interface First extends fakePackage1.Interface {}
+export interface Second extends fakePackage2.Interface {}
+export interface Third extends fakePackage3.Interface {}
+
+export { ExposedInterface };

--- a/tests/e2e/test-cases/import-with-different-names/other-module.ts
+++ b/tests/e2e/test-cases/import-with-different-names/other-module.ts
@@ -1,0 +1,6 @@
+import fakePackageDifferentNameThatShouldNotBeAddedToOutput = require('fake-package');
+import PackageWithExportEq = require('package-with-export-eq');
+
+export interface OtherInternalInterface extends fakePackageDifferentNameThatShouldNotBeAddedToOutput.Interface {}
+
+export interface ExposedInterface extends PackageWithExportEq.SomeCoolInterface {}

--- a/tests/e2e/test-cases/import-with-different-names/output.d.ts
+++ b/tests/e2e/test-cases/import-with-different-names/output.d.ts
@@ -1,0 +1,15 @@
+import * as fakePackage2 from 'fake-package';
+import fakePackage3 = require('fake-package');
+import fakePackage1 from 'fake-package';
+import PackageWithExportEq = require('package-with-export-eq');
+
+export interface ExposedInterface extends PackageWithExportEq.SomeCoolInterface {
+}
+export interface First extends fakePackage1.Interface {
+}
+export interface Second extends fakePackage2.Interface {
+}
+export interface Third extends fakePackage3.Interface {
+}
+
+export {};

--- a/tests/e2e/test-cases/import-with-different-names/tsconfig.json
+++ b/tests/e2e/test-cases/import-with-different-names/tsconfig.json
@@ -1,0 +1,7 @@
+{
+	"extends": "../tsconfig.json",
+	"compilerOptions": {
+		"allowSyntheticDefaultImports": true,
+		"esModuleInterop": true
+	}
+}

